### PR TITLE
esp32: Add board config to support MCP2515

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add board config to support MCP2515
## Impact
Only ESP32-Devkit
## Testing
ESP32-Devkit

### CODE REVIEW

File: boards/xtensa/esp32/esp32-devkitc/src/esp32_mcp2515.c [hunk 1]
```diff
@@ -0,0 +1,250 @@
+/****************************************************************************
+ * boards/xtensa/esp32/esp32devkitc/src/esp32_mcp2515.c
```
Comment (Ouss4) [inline boards/xtensa/esp32/esp32-devkitc/src/esp32_mcp2515.c:2]: Why is this not part of the common board logic?
I checked stm32 and it also there only stm32f3-minimum and not in the common folder.  Is there any particularities?  I can't see any from my side.

Comment (acassis) [reply boards/xtensa/esp32/esp32-devkitc/src/esp32_mcp2515.c:2]: you are right, it could become a common board logic, I will modify it!

**Comment by acassis** [comment]
Hi @Ouss4 I fixed the suggestion. He build succeed, only a MacOS failed and sometimes it does. Could you please merge it?

**Review by Ouss4 — APPROVED** [review]
LGTM

**Pull request merged**